solvation_analysis/analysis.py

```diff
@@ -0,0 +1,321 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.analysis.rdf import InterRDF
+from MDAnalysis.lib.distances import capped_distance
+import numpy as np
+from solvation_analysis.rdf_parser import identify_solvation_cutoff
+from solvation_analysis.analysis_library import (
+    _Coordination,
+    _Pairing,
+    _Speciation,
+)
+from solvation_analysis.solvation import get_radial_shell, get_closest_n_mol, get_atom_group
+
+
+class Solution(AnalysisBase):
+    """
+    The core class of the solvation module.
+    """
+
+    def __init__(
+        self,
+        solute,
+        solvents,
+        radii=None,
+        rdf_kernel=None,
+        kernel_kwargs=None,
+        rdf_init_kwargs=None,
+        rdf_run_kwargs=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+            solute : AtomGroup
+                the solute in the solutions
+            solvents: dict
+                a dictionary of names and atom groups. e.g. {"name_1": solvent_group_1,
+                "name_2": solvent_group_2, ...}
+            radii : dict
+                an optional dictionary of solvation radii, any radii not
+                given will be calculated. e.g. {"name_2": radius_2, "name_5": radius_5}
+            rdf_kernel : function
+                this function must take rdf bins and data as input and return
+                a solvation radius as output. e.g. rdf_kernel(bins, data) -> 3.2. By default,
+                the rdf_kernel is solvation_analysis.rdf_parser.identify_solvation_cutoff.
+            kernel_kwargs : dict
+                kwargs passed to rdf_kernel
+            rdf_init_kwargs : dict
+                kwargs passed to inner rdf initialization
+            rdf_run_kwargs : dict
+                kwargs passed to inner rdf run e.g. inner_rdf.run(**rdf_run_kwargs)
+            kwargs : dict
+                kwargs passed to AnalysisBase
+        """
+        super(Solution, self).__init__(solute.universe.trajectory, **kwargs)
```

Comment:
You can add a numpy docstrings section that describes the attributes of the class, I think that would be a big help in this case.

Reply:
Many of the parameters are directly saved as attributes. Should I include those arguments in both the attributes and parameters sections of the docs?

Reply:
Nah just in parameters, attributes is for a datastructure that is only present once the analysis has been run in this case.